java.lang.ClassNotFoundException: jakarta.ws.rs.core.Configuration

**SpringBoot: 2.7.2
smart-id-java-client: 2.2**

Caused by: java.lang.IllegalStateException: Failed to introspect Class [ee.sk.smartid.SmartIdClient] from ClassLoader [jdk.internal.loader.ClassLoaders$AppClassLoader@73d16e93]
        at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:485)
        at org.springframework.util.ReflectionUtils.doWithLocalMethods(ReflectionUtils.java:321)
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.determineCandidateConstructors(AutowiredAnnotationBeanPostProcessor.java:267)
        ... 32 common frames omitted
Caused by: java.lang.NoClassDefFoundError: jakarta/ws/rs/core/Configuration
        at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
        at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3325)
        at java.base/java.lang.Class.getDeclaredMethods(Class.java:2466)
        at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:467)
        ... 34 common frames omitted
Caused by: java.lang.ClassNotFoundException: jakarta.ws.rs.core.Configuration
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:606)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:168)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 38 common frames omitted

**Fix:** requires extra dependency jakarta.ws.rs:jakarta.ws.rs-api to overcome the problem!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## [Unreleased] 
 
+### Fixed
+- added jakarta.ws.rs:jakarta.ws.rs-api as a dependency to avoid ClassNotFoundException with spring framework
+
 ### Changed
 - Updated dependencies
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,14 @@
 All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+## [Unreleased] 
+
+### Changed
+- Updated dependencies
+
+### Changes in tests and documentation
+- How to use a proxy server - added documentation to README.md and tests to ReadmeTest.java
+
 ## [2.2] - 2022-02-22
 
 ### Changed

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,7 +2,7 @@
 All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
-## [Unreleased] 
+## [2.2.1] - 2022-09-12
 
 ### Fixed
 - added jakarta.ws.rs:jakarta.ws.rs-api as a dependency to avoid ClassNotFoundException with spring framework

--- a/README.md
+++ b/README.md
@@ -18,33 +18,34 @@ For using Smart-ID API v. 1.0 see [Smart-ID Java Client 1.X](https://github.com/
     *   [Requirements](#requirements)
     *   [Getting the library](#getting-the-library)
     *   [Changelog](#changelog)
-*  [How to use it](#how-to-use-it)
-    *   [Test accounts for testing]()
-    *   [Logging](#logging)
+* [How to use it](#how-to-use-it)
+    * [Test accounts for testing]()
+    * [Logging](#logging)
         *   [Log request payloads](#log-request-payloads)
         *   [Get the IP address of user's device](#get-the-ip-address-of-users-device)
-    *   [Example of configuring the client](#example-of-configuring-the-client)
+    * [Example of configuring the client](#example-of-configuring-the-client)
         *   [Reading trusted certificates from key store](#reading-trusted-certificates-from-key-store)
         *   [Feeding trusted certificates one by one](#feeding-trusted-certificates-one-by-one)
-    *   [Examples of performing authentication](#examples-of-performing-authentication)
+    * [Examples of performing authentication](#examples-of-performing-authentication)
         *   [Authenticating with semantics identifier](#authenticating-with-semantics-identifier)
         *   [Authenticating with document number](#authenticating-with-document-number)
         *   [Validating authentication response](#validating-authentication-response)
-    *   [Creating a signature](#creating-a-signature)
+    * [Creating a signature](#creating-a-signature)
         *   [Obtaining signer's certificate](#obtaining-signers-certificate)
         *   [Create the signature](#create-the-signature)
-    *   [Setting the order of preferred interactions for displaying text and asking PIN](#setting-the-order-of-preferred-interactions-for-displaying-text-and-asking-pin)
+    * [Setting the order of preferred interactions for displaying text and asking PIN](#setting-the-order-of-preferred-interactions-for-displaying-text-and-asking-pin)
         *   [Parameter allowedInteractionsOrder most common examples](#parameter-allowedinteractionsorder-most-common-examples)
             *   [Short confirmation message with PIN](#short-confirmation-message-with-pin)
             *   [Verification code choice](#verification-code-choice)
             *   [Long confirmation message with fallback to PIN](#long-confirmation-message-with-fallback-to-pin)
             *   [Long confirmation message together with verification code choice with fallback to verification code choice](#long-confirmation-message-together-with-verification-code-choice-with-fallback-to-verification-code-choice)
             *   [Interactions with longer text without fallback](#interactions-with-longer-text-without-fallback)
-    *   [Handling exceptions](#handling-exceptions)
-    *   [Network connection configuration of the client](#network-connection-configuration-of-the-client)
+    * [Handling exceptions](#handling-exceptions)
+    * [Network connection configuration of the client](#network-connection-configuration-of-the-client)
         *   [Example of creating a client with configured ssl context on JBoss using JAXWS RS](#example-of-creating-a-client-with-configured-ssl-context-on-jboss-using-jaxws-rs)
-        *   [Example of creating a client with configured proxy on JBoss](#example-of-creating-a-client-with-configured-ssl-context-on-jboss-using-jaxws-rs)
-
+    * [Configuring a proxy](#configuring-a-proxy)
+        * [Configuring a proxy using JBoss Resteasy library](#configuring-a-proxy-using-jboss-resteasy-library)
+        * [Configuring a proxy using Jersey](#configuring-a-proxy-using-jersey)
 
 ## Introduction
 
@@ -560,19 +561,40 @@ client.setHostUrl("https://sid.demo.sk.ee/smart-id-rp/v2/");
 client.setConfiguredClient(resteasyClient);
 ```
 
+## Configuring a proxy
+
+If you need to access the internet through a proxy (that runs on 127.0.0.1:3128 in the examples)
+you have two alternatives:
+
+### Configuring a proxy using JBoss Resteasy library
+
+<!-- Do not change code samples here but instead copy from ReadmeTest.document_setProxy_withJbossRestEasy() -->
+```java   
+    org.jboss.resteasy.client.jaxrs.ResteasyClient resteasyClient =
+            new org.jboss.resteasy.client.jaxrs.internal.ResteasyClientBuilderImpl()
+                    .defaultProxy("127.0.0.1", 3128, "http")
+                    .build();
+    SmartIdClient client = new SmartIdClient();
+    client.setRelyingPartyUUID("00000000-0000-0000-0000-000000000000");
+    client.setRelyingPartyName("DEMO");
+    client.setHostUrl("https://sid.demo.sk.ee/smart-id-rp/v2/");
+    client.setConfiguredClient(resteasyClient);
+    client.setTrustedCertificates(DEMO_HOST_SSL_CERTIFICATE);
+```
 
 ### Example of creating a client with configured proxy on JBoss
 
-```java   
-ResteasyClient resteasyClient = new ResteasyClientBuilder()
-        .defaultProxy("localhost", 8080, "http")
-        .build();
+<!-- Do not change code samples here but instead copy from ReadmeTest.document_setNetworkConnectionConfig_withJersey()-->
+```java 
+    org.glassfish.jersey.client.ClientConfig clientConfig =
+            new org.glassfish.jersey.client.ClientConfig();
+    clientConfig.property(ClientProperties.PROXY_URI, "http://127.0.0.1:3128");
 
-SmartIdClient client = new SmartIdClient();
-client.setRelyingPartyUUID("00000000-0000-0000-0000-000000000000");
-client.setRelyingPartyName("DEMO");
-client.setHostUrl("https://sid.demo.sk.ee/smart-id-rp/v2/");
-client.setConfiguredClient(resteasyClient);
+    SmartIdClient client = new SmartIdClient();
+    client.setRelyingPartyUUID("00000000-0000-0000-0000-000000000000");
+    client.setRelyingPartyName("DEMO");
+    client.setHostUrl("https://sid.demo.sk.ee/smart-id-rp/v2/");
+    client.setNetworkConnectionConfig(clientConfig);
+    client.setTrustedCertificates(DEMO_HOST_SSL_CERTIFICATE);
 ```
 
-    

--- a/pom.xml
+++ b/pom.xml
@@ -47,6 +47,7 @@
         <maven.compiler.target>1.8</maven.compiler.target>
         <jackson.version>2.13.3</jackson.version>
         <jersey.version>3.0.6</jersey.version>
+        <rs-api.version>3.0.0</rs-api.version>
         <resteasy.version>6.0.3.Final</resteasy.version>
     </properties>
 
@@ -66,6 +67,13 @@
             <artifactId>jersey-hk2</artifactId>
             <version>${jersey.version}</version>
         </dependency>
+
+        <dependency>
+            <groupId>jakarta.ws.rs</groupId>
+            <artifactId>jakarta.ws.rs-api</artifactId>
+            <version>${rs-api.version}</version>
+        </dependency>
+
         <dependency>
             <groupId>org.glassfish.jersey.connectors</groupId>
             <artifactId>jersey-apache-connector</artifactId>

--- a/pom.xml
+++ b/pom.xml
@@ -45,8 +45,9 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
-        <jackson.version>2.13.1</jackson.version>
-        <jersey.version>3.0.4</jersey.version>
+        <jackson.version>2.13.3</jackson.version>
+        <jersey.version>3.0.6</jersey.version>
+        <resteasy.version>6.0.3.Final</resteasy.version>
     </properties>
 
     <dependencies>
@@ -93,8 +94,16 @@
         <dependency>
             <groupId>org.glassfish.jaxb</groupId>
             <artifactId>jaxb-runtime</artifactId>
-            <version>3.0.1</version>
+            <version>4.0.0</version>
         </dependency>
+
+
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcprov-jdk15on</artifactId>
+            <version>1.70</version>
+        </dependency>
+
 
         <dependency>
             <groupId>junit</groupId>
@@ -111,38 +120,35 @@
         <dependency>
             <groupId>ch.qos.logback</groupId>
             <artifactId>logback-classic</artifactId>
-            <version>1.2.10</version>
+            <version>1.2.11</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>com.github.tomakehurst</groupId>
             <artifactId>wiremock</artifactId>
-            <version>2.4.1</version>
+            <version>2.27.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>4.3.1</version>
+            <version>4.7.0</version>
             <scope>test</scope>
         </dependency>
 
-
-        <dependency>
-            <groupId>org.bouncycastle</groupId>
-            <artifactId>bcprov-jdk15on</artifactId>
-            <version>1.69</version>
-        </dependency>
-
-        <!-- comment in if you want to configure client with RestEasy in tests
         <dependency>
             <groupId>org.jboss.resteasy</groupId>
             <artifactId>resteasy-client</artifactId>
-            <version>3.12.0.Final</version>
+            <version>${resteasy.version}</version>
             <scope>test</scope>
-            <optional>true</optional>
         </dependency>
-        -->
+
+        <dependency>
+            <groupId>org.jboss.resteasy</groupId>
+            <artifactId>resteasy-jackson2-provider</artifactId>
+            <version>${resteasy.version}</version>
+            <scope>test</scope>
+        </dependency>
 
     </dependencies>
 

--- a/src/test/java/ee/sk/smartid/rest/SmartIdRestIntegrationTest.java
+++ b/src/test/java/ee/sk/smartid/rest/SmartIdRestIntegrationTest.java
@@ -66,12 +66,12 @@ public class SmartIdRestIntegrationTest {
   public void getCertificateAndSignHash() throws Exception {
     CertificateChoiceResponse certificateChoiceResponse = fetchCertificateChoiceSession(DOCUMENT_NUMBER_LT);
 
-    SessionStatus sessionStatus = pollSessionStatus(certificateChoiceResponse.getSessionID());
+    SessionStatus sessionStatus = pollSessionStatus(certificateChoiceResponse.getSessionID(), connector);
     assertCertificateChosen(sessionStatus);
 
     String documentNumber = sessionStatus.getResult().getDocumentNumber();
     SignatureSessionResponse signatureSessionResponse = createRequestAndFetchSignatureSession(documentNumber);
-    sessionStatus = pollSessionStatus(signatureSessionResponse.getSessionID());
+    sessionStatus = pollSessionStatus(signatureSessionResponse.getSessionID(), connector);
     assertSignatureCreated(sessionStatus);
   }
 
@@ -85,7 +85,7 @@ public class SmartIdRestIntegrationTest {
     assertNotNull(authenticationSessionResponse);
     assertThat(authenticationSessionResponse.getSessionID(), not(isEmptyOrNullString()));
 
-    SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID());
+    SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID(), connector);
     assertAuthenticationResponseCreated(sessionStatus);
   }
 
@@ -97,7 +97,7 @@ public class SmartIdRestIntegrationTest {
     assertNotNull(authenticationSessionResponse);
     assertThat(authenticationSessionResponse.getSessionID(), not(isEmptyOrNullString()));
 
-    SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID());
+    SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID(), connector);
 
     assertNotNull(sessionStatus.getResult());
     assertThat(sessionStatus.getResult().getEndResult(), is("OK"));
@@ -124,7 +124,7 @@ public class SmartIdRestIntegrationTest {
     assertNotNull(authenticationSessionResponse);
     assertThat(authenticationSessionResponse.getSessionID(), not(isEmptyOrNullString()));
 
-    SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID());
+    SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID(), connector);
 
     assertNotNull(sessionStatus.getResult());
     assertThat(sessionStatus.getResult().getEndResult(), is("OK"));
@@ -137,7 +137,7 @@ public class SmartIdRestIntegrationTest {
   public void getIgnoredProperties_withSign_getIgnoredProperties_withAuthenticate_testAccountsIgnoreVcChoice() throws Exception {
     CertificateChoiceResponse certificateChoiceResponse = fetchCertificateChoiceSession(DOCUMENT_NUMBER);
 
-    SessionStatus sessionStatus = pollSessionStatus(certificateChoiceResponse.getSessionID());
+    SessionStatus sessionStatus = pollSessionStatus(certificateChoiceResponse.getSessionID(), connector);
     assertCertificateChosen(sessionStatus);
 
     String documentNumber = sessionStatus.getResult().getDocumentNumber();
@@ -145,7 +145,7 @@ public class SmartIdRestIntegrationTest {
     SignatureSessionRequest signatureSessionRequest = createSignatureSessionRequest();
 
     SignatureSessionResponse signatureSessionResponse = fetchSignatureSession(documentNumber, signatureSessionRequest);
-    sessionStatus = pollSessionStatus(signatureSessionResponse.getSessionID());
+    sessionStatus = pollSessionStatus(signatureSessionResponse.getSessionID(), connector);
 
     assertNotNull(sessionStatus.getResult());
     assertThat(sessionStatus.getResult().getEndResult(), is("OK"));
@@ -172,7 +172,7 @@ public class SmartIdRestIntegrationTest {
     assertNotNull(authenticationSessionResponse);
     assertThat(authenticationSessionResponse.getSessionID(), not(isEmptyOrNullString()));
 
-    SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID());
+    SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID(), connector);
 
     assertThat(sessionStatus.getInteractionFlowUsed(), is("displayTextAndPIN"));
 
@@ -221,7 +221,7 @@ public class SmartIdRestIntegrationTest {
     return signatureSessionRequest;
   }
 
-  private AuthenticationSessionRequest createAuthenticationSessionRequest() {
+  public static AuthenticationSessionRequest createAuthenticationSessionRequest() {
     AuthenticationSessionRequest authenticationSessionRequest = new AuthenticationSessionRequest();
     authenticationSessionRequest.setRelyingPartyUUID(RELYING_PARTY_UUID);
     authenticationSessionRequest.setRelyingPartyName(RELYING_PARTY_NAME);
@@ -235,10 +235,10 @@ public class SmartIdRestIntegrationTest {
     return authenticationSessionRequest;
   }
 
-  private SessionStatus pollSessionStatus(String sessionId) throws InterruptedException {
+  public static SessionStatus pollSessionStatus(String sessionId, SmartIdConnector connector1) throws InterruptedException {
     SessionStatus sessionStatus = null;
     while (sessionStatus == null || "RUNNING".equalsIgnoreCase(sessionStatus.getState() )) {
-      sessionStatus = connector.getSessionStatus(sessionId);
+      sessionStatus = connector1.getSessionStatus(sessionId);
       TimeUnit.SECONDS.sleep(1);
     }
     assertEquals("COMPLETE", sessionStatus.getState());
@@ -258,7 +258,7 @@ public class SmartIdRestIntegrationTest {
     assertThat(sessionStatus.getCert().getValue(), not(isEmptyOrNullString()));
   }
 
-  private void assertAuthenticationResponseCreated(SessionStatus sessionStatus) {
+  public static void assertAuthenticationResponseCreated(SessionStatus sessionStatus) {
     assertNotNull(sessionStatus);
 
     assertThat(sessionStatus.getResult().getEndResult(), not(isEmptyOrNullString()));
@@ -267,7 +267,7 @@ public class SmartIdRestIntegrationTest {
     assertThat(sessionStatus.getCert().getCertificateLevel(), not(isEmptyOrNullString()));
   }
 
-  private String calculateHashInBase64(byte[] dataToSign) {
+  private static String calculateHashInBase64(byte[] dataToSign) {
     byte[] digestValue = DigestCalculator.calculateDigest(dataToSign, HashType.SHA512);
     return Base64.encodeBase64String(digestValue);
   }

--- a/src/test/java/ee/sk/test/smartid/integration/ReadmeTest.java
+++ b/src/test/java/ee/sk/test/smartid/integration/ReadmeTest.java
@@ -31,15 +31,15 @@ import ee.sk.smartid.exception.UnprocessableSmartIdResponseException;
 import ee.sk.smartid.exception.permanent.SmartIdClientException;
 import ee.sk.smartid.exception.useraccount.RequiredInteractionNotSupportedByAppException;
 import ee.sk.smartid.exception.useraction.UserSelectedWrongVerificationCodeException;
-import ee.sk.smartid.rest.dao.Interaction;
-import ee.sk.smartid.rest.dao.InteractionFlow;
-import ee.sk.smartid.rest.dao.SemanticsIdentifier;
+import ee.sk.smartid.rest.SmartIdConnector;
+import ee.sk.smartid.rest.dao.*;
 import org.apache.http.client.config.RequestConfig;
 import org.glassfish.jersey.apache.connector.ApacheClientProperties;
 import org.glassfish.jersey.apache.connector.ApacheConnectorProvider;
 import org.glassfish.jersey.client.ClientConfig;
 import org.glassfish.jersey.client.ClientProperties;
 import org.junit.Before;
+import org.junit.Ignore;
 import org.junit.Test;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -53,8 +53,13 @@ import java.util.Collections;
 import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 
+import static ee.sk.smartid.rest.SmartIdRestIntegrationTest.*;
 import static ee.sk.test.smartid.integration.SmartIdIntegrationTest.TEST_AGAINST_SMART_ID_DEMO;
 import static java.util.Arrays.asList;
+import static junit.framework.TestCase.assertNotNull;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.not;
 import static org.junit.Assume.assumeTrue;
 
 /**
@@ -622,61 +627,83 @@ Here's an example how to configure HTTP connector's custom socket timeouts for t
         client.setNetworkConnectionConfig(clientConfig);
     }
 
-    /*
-
-    Keep in mind that the HTTP connector timeout of waiting for data shouldn't normally be less than the timeout for session status poll.
-
-    ### Example of creating a client with configured ssl context on JBoss using JAXWS RS
-
-
-
-     */
-
-
 
     @Test
-    public void documentJbossJaxWS() {
+    @Ignore("you need to run a proxy to run this test")
+    public void document_setProxy_withJbossRestEasy() throws Exception {
+        // in order to run this test you can set up a proxy server locally
+        //docker run -d --name squid-container -e TZ=UTC -p 3128:3128 ubuntu/squid:5.2-22.04_beta
 
-        /*
-        ResteasyClient resteasyClient = new ResteasyClientBuilder()
-                .sslContext(SmartIdClient.createSslContext(Arrays.asList("pem cert 1", "pem cert 2")))
-                .build();
 
+        // CODE EXAMPLE STARTS HERE
+
+        org.jboss.resteasy.client.jaxrs.ResteasyClient resteasyClient =
+                new org.jboss.resteasy.client.jaxrs.internal.ResteasyClientBuilderImpl()
+                        .defaultProxy("127.0.0.1", 3128, "http")
+                        .build();
         SmartIdClient client = new SmartIdClient();
         client.setRelyingPartyUUID("00000000-0000-0000-0000-000000000000");
         client.setRelyingPartyName("DEMO");
         client.setHostUrl("https://sid.demo.sk.ee/smart-id-rp/v2/");
         client.setConfiguredClient(resteasyClient);
-*/
+        client.setTrustedCertificates(DEMO_HOST_SSL_CERTIFICATE);
 
+        // CODE EXAMPLE ENDS HERE
+
+
+        SemanticsIdentifier semanticsIdentifier = new SemanticsIdentifier(SemanticsIdentifier.IdentityType.PNO, SemanticsIdentifier.CountryCode.LV, "030303-10012");
+
+        AuthenticationSessionRequest request =  createAuthenticationSessionRequest();
+        SmartIdConnector smartIdConnector = client.getSmartIdConnector();
+        AuthenticationSessionResponse authenticationSessionResponse = smartIdConnector.authenticate(semanticsIdentifier, request);
+
+        assertNotNull(authenticationSessionResponse);
+        assertThat(authenticationSessionResponse.getSessionID(), not(isEmptyOrNullString()));
+
+        SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID(), smartIdConnector);
+        assertAuthenticationResponseCreated(sessionStatus);
+
+
+        // this allows to switch off tests going against smart-id demo env
+        assumeTrue(TEST_AGAINST_SMART_ID_DEMO);
     }
 
-    /*
-
-    ### Example of creating a client with configured proxy on JBoss
-
-
-     */
-
     @Test
-    public void configureClientWithNetwork() {
+    @Ignore("you need a running proxy server to run this test")
+    public void document_setNetworkConnectionConfig_withJersey() throws Exception {
+        // in order to run this test you first have to set up a proxy server locally
+        //docker run -d --name squid-container -e TZ=UTC -p 3128:3128 ubuntu/squid:5.2-22.04_beta
 
-        /* To comment in this you need to enable JBoss repository and dependency in pom.xml
+        // CODE EXAMPLE STARTS HERE
 
-        ResteasyClient resteasyClient = new ResteasyClientBuilder()
-                .defaultProxy("localhost", 8080, "http")
-                .build();
+        org.glassfish.jersey.client.ClientConfig clientConfig =
+                new org.glassfish.jersey.client.ClientConfig();
+        clientConfig.property(ClientProperties.PROXY_URI, "http://127.0.0.1:3128");
 
         SmartIdClient client = new SmartIdClient();
         client.setRelyingPartyUUID("00000000-0000-0000-0000-000000000000");
         client.setRelyingPartyName("DEMO");
         client.setHostUrl("https://sid.demo.sk.ee/smart-id-rp/v2/");
-        client.setConfiguredClient(resteasyClient);
+        client.setNetworkConnectionConfig(clientConfig);
+        client.setTrustedCertificates(DEMO_HOST_SSL_CERTIFICATE);
 
-         */
+        // CODE EXAMPLE ENDS HERE
+
+        SemanticsIdentifier semanticsIdentifier = new SemanticsIdentifier(SemanticsIdentifier.IdentityType.PNO, SemanticsIdentifier.CountryCode.LV, "030303-10012");
+
+        AuthenticationSessionRequest request =  createAuthenticationSessionRequest();
+        SmartIdConnector smartIdConnector = client.getSmartIdConnector();
+        AuthenticationSessionResponse authenticationSessionResponse = smartIdConnector.authenticate(semanticsIdentifier, request);
+
+        assertNotNull(authenticationSessionResponse);
+        assertThat(authenticationSessionResponse.getSessionID(), not(isEmptyOrNullString()));
+
+        SessionStatus sessionStatus = pollSessionStatus(authenticationSessionResponse.getSessionID(), smartIdConnector);
+        assertAuthenticationResponseCreated(sessionStatus);
+
+        // this allows to switch off tests going against smart-id demo env
+        assumeTrue(TEST_AGAINST_SMART_ID_DEMO);
+
     }
-
-
-
 
 }

